packages/web-components/fast-element/src/templating/html-directive.ts

```diff
@@ -87,13 +87,67 @@ export abstract class HTMLDirective implements ViewBehaviorFactory {
     public abstract createBehavior(targets: ViewBehaviorTargets): Behavior | ViewBehavior;
 }
 
+/**
+ * The type of HTML aspect to target.
+ */
+export enum Aspect {
+    /**
+     * An attribute.
+     */
+    attribute = 0,
+    /**
+     * A boolean attribute.
+     */
+    booleanAttribute = 1,
+    /**
+     * A property.
+     */
+    property = 2,
+    /**
+     * Content
+     */
+    content = 3,
+    /**
+     * A token list.
+     */
+    tokenList = 4,
```

Comment:
What is a token list? Will this need to be handled in SSR infrastructure?

Reply:
This is used for special handling of CSS classes. In v2, by default, when you add a binding to `class` it will now use a property binding against `className` for better performance. However, you can also use an explicit property binding on `classList` which will use a special token list binding behavior that attempts to sync class add/removal across various sources. In other words, this handles changes to the class list that happen from outside the binding in addition to handling the binding. It's more costly and only needed in certain cases, most notable, host class bindings (which I think we're trying to avoid in most of our scenarios anyway).

For SSR, I don't think you need to differentiate these cases.